### PR TITLE
Lending added

### PR DESCRIPTION
**Pull request by akshay111meher**

### CODE REVIEW

File: contracts/Pool/Pool.sol [hunk 1]
```diff
@@ -16,6 +14,8 @@ import "../interfaces/IPool.sol";
 import "../interfaces/IExtension.sol";
 import "../interfaces/IPoolToken.sol";
 
+import "hardhat/console.sol";
```
Comment (prateekreddy) [inline contracts/Pool/Pool.sol:17]: Can you remove this ?

File: contracts/Pool/Pool.sol [hunk 2]
```diff
@@ -275,12 +275,19 @@ contract Pool is Initializable, IPool, ReentrancyGuard {
...
+                //     _poolSavingsStrategy,
+                //     _liquidityshare
+                // );
                 _sharesReceived = _savingsAccount.transferFrom(
                     _asset,
                     _depositFrom,
                     address(this),
                     _poolSavingsStrategy,
-                    _liquidityshare
+                    _amount
```
Comment (prateekreddy) [inline contracts/Pool/Pool.sol:290]: Why this change ?

File: contracts/Pool/Pool.sol [hunk 3]
```diff
@@ -275,12 +275,19 @@ contract Pool is Initializable, IPool, ReentrancyGuard {
                         _asset
                     );
             if (_toSavingsAccount) {
+                // _sharesReceived = _savingsAccount.transferFrom(
+                //     _asset,
+                //     _depositFrom,
+                //     address(this),
+                //     _poolSavingsStrategy,
+                //     _liquidityshare
+                // );
```
Comment (prateekreddy) [inline contracts/Pool/Pool.sol:284]: Don't comment, remove completely.

**Review by prateekreddy — CHANGES_REQUESTED** [review]

File: contracts/Pool/Pool.sol (hunk 2 above)
Comment (akshay111meher) [reply contracts/Pool/Pool.sol:290]: liquidity shares seem to be a wrong input there. 

When consoled, _amount was making the transaction pass. 

Tried simulating with some tokens as well.

File: contracts/Pool/Pool.sol (hunk 3 above)
Comment (akshay111meher) [reply contracts/Pool/Pool.sol:284]: Will resolve this once, the above comment is confirmed..

File: contracts/Pool/Pool.sol (hunk 2 above)
Comment (prateekreddy) [reply contracts/Pool/Pool.sol:288]: Makes sense, savings account should  take actually value as input, not the liquidity shares

**Comment by akshay111meher** [comment]
The merge is redundant. 
Inclusive changes in other branch

**Pull request closed without merging**